src/components/tests/loop-form.test.tsx

```diff
@@ -15,59 +14,58 @@ const buildWrapper = (listOfPlaces?: string[]) => {
     console.log('test');
   };
 
-  const wrapper = mount(
-    <LoopForm
-      requestLoop={onClick}
-      fleets={['SuperFleet']}
-      listOfPlaces={!!listOfPlaces ? listOfPlaces : fakePlaces()['SuperFleet']}
-    />,
-  );
+  const wrapper = mount(<LoopForm requestLoop={onClick} fleets={[fleetName]} />);
   return wrapper;
 };
 
 describe('form Validation', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
   test('Initial Values', () => {
-    const wrapper = buildWrapper();
+    const wrapper = buildWrapper('SuperFleet');
+    expect(wrapper.find('input[name="targetFleet"]').props().value != '');
     expect(wrapper.find("input[type='number']").props().value).toEqual('');
-    expect(wrapper.findWhere(x => x.name() === 'input' && x.props().value === 'supplies'))
-      .toBeTruthy;
-    expect(
-      wrapper.findWhere(
-        x => x.name() === 'input' && x.props().value === fakePlaces()['SuperFleet'][0],
-      ),
-    ).toBeTruthy;
-
-    expect(
-      wrapper.findWhere(
-        x => x.name() === 'input' && x.props().value === fakePlaces()['SuperFleet'][1],
-      ),
-    ).toBeTruthy;
+    expect(wrapper.findWhere(x => x.name() === 'input' && x.props().value != '')).toBeTruthy();
+    expect(wrapper.find('input[name="startLocation"]').props().value).not.toEqual(
+      wrapper.find('input[name="finishLocation"]').props().value,
+    );
     wrapper.unmount();
   });
 
+  test('Successful Request', () => {
+    const spy = jest.spyOn(console, 'log');
+    const wrapper = buildWrapper('SuperFleet');
+    wrapper.find("input[type='number']").simulate('change', { target: { value: 1 } });
+    expect(wrapper.find("input[type='number']").props().value).toEqual(1);
+    wrapper.find('form').simulate('submit');
+    expect(spy).toBeCalled();
+  });
+
   test('Number of loops cannot be empty', async () => {
-    const wrapper = buildWrapper();
+    const spy = jest.spyOn(console, 'log');
+    const wrapper = buildWrapper('SuperFleet');
     wrapper.find('form').simulate('submit');
-    expect(wrapper.find('#numLoopsError').html()).toContain('Loops can only be &gt; 0');
+    expect(spy).not.toBeCalled();
+    expect(wrapper.exists('.Mui-error')).toBeTruthy();
     wrapper.unmount();
   });
 
   test('Location cannot be empty', async () => {
-    const wrapper = buildWrapper(['placeA']);
+    const spy = jest.spyOn(console, 'log');
+    const wrapper = buildWrapper('FleetA');
     wrapper.find('form').simulate('submit');
-    expect(wrapper.find('#startLocationError').html()).toContain('Location cannot be empty');
+    expect(spy).not.toBeCalled();
+    expect(wrapper.exists('.Mui-error')).toBeTruthy();
     wrapper.unmount();
   });
 
-  test('Start Location cannot be equal to finish Location', async () => {
-    const wrapper = buildWrapper(['placeA', 'placeA', 'placeA']);
+  test('Start Location cannot be equal to Finish Location', async () => {
+    const spy = jest.spyOn(console, 'log');
```

Comment:
This seems like a questionable spy. console.log can be called pretty much anywhere.

Reply:
@koonpeng Do you have any suggestion regarding this?
what I can think of is using `expect(spy).toBeCalledWith('test')` / `expect(spy).not.toBeCalledWith('test')` and the onClick event handler could produce a more specific console.log statement like 'testing loop form'
```
const onClick = (
    fleetName: string,
    numLoops: number,
    startLocationPoint: string,
    endLocationPoint: string,
  ) => {
    console.log('testing the loop form');
  };
...
expect(spy).toBeCalledWith('testing the loop form');
```
@matiasbavera if you have any suggestion for this, please do chime in! (:

Reply:
How about something like this?

``` typescript
test('Start Location cannot be equal to Finish Location', async () => {
  let clicked = false;
  const onClick = (
    fleetName: string,
    numLoops: number,
    startLocationPoint: string,
    endLocationPoint: string,
  ) => {
    clicked = true;
  };
  // You can send a custom onClick function to your buildWrapper function
  const wrapper = buildWrapper('FleetB', onClick);
  wrapper.find('form').simulate('submit');
  expect(clicked).toBeFalsy();
  /**
   * Also it should be good to check for the exact error. You can do that setting an Id to the Autocomplete component and accessing to that html element in this way .find('#<ID>-helper-text')
   * You can check an example here https://github.com/osrf/romi-dashboard/blob/2d22429068d014bc7257b0a28cbe8d0fa2c93498/src/components/tests/robot-item-delivery-form.test.tsx
   */
  expect(wrapper.exists('.Mui-error')).toBeTruthy();
  wrapper.unmount();
});

```